ISISComputingGroup/ibex_gui
# Ticket931 pinhole opi alterations

The pinhole should just look a little cleaner

Set up for testing:
Make sure that the system you are using has a motion set points version from January 2016 (post merge #1, note that when writing Demo had not been updated)
Make sure that the system you are using is set up to simulate Galils (globals.txt at instrument level in instrument/settings/config/[inst-name] - see Demo for an example of the file)
Make sure that the system you are using has the directory instrument/settings/config/[inst-name]/galil, with a motionsetpoints.cmd which includes the pinhole selector information (see Demo for an example)
Make sure that the system you are using has the directory instrument/settings/config/[inst-name]/motionSetPoints which has an entry for pinhole_selector.txt with 12 entries (see Demo for an example)
Create a synoptic with a pinhole selector listed on it, set the macros to be PH=PINHOLE and MM to be the motor specified in the cmd files (MTR0601 in the files on Demo when writing)
Make sure that the appropriate Galil is running (e.g. GALIL_06 if using the values as per Demo)

To test:
Open the OPI from the synoptic
The drop down should set to "Running", and the buttons should read in the format "A=position name from pinhole_selector.txt" (If only the letters are available then the version of motion set points is out of date)
If the position is one of the running positions then the pinhole graphic should have the appropriate letter in the top position, with the other letters continuing around the edge. If the position is one of the loading positions, then the graphic will show only question marks.
Press a button, and the simulated system should move around to put the selected letter into the active position (highlighted by the red square). It will show questions marks during this move, as it will pass through loading positions whilst moving, as well as tracking the positions it passes through.
Try a few different positions.
Select Loading from the drop down.
The graphic should change to question marks in a slightly different configuration, with the red square now being in the '9 o'clock' position.
The buttons should have re-labelled to the loading names.
Again pressing buttons should move the system, with the graphic updating.

The other items on the screen were not altered, however, the Reload Positions File should also update the names on the buttons, so it might be worth altering the labels in the pinhole_selector.txt file and pressing this button to make sure that the new labels are now being used.

(Note, the set up of the motor might mean that only certain positions are allowed)
